File: DictionaryProject/Commands/RepeatCommand.vb
<Highlight(RegEx:="^\s*repeat$", Color:="Green")> _
Public Class RepeatCommand
    Inherits CommandBase

    Public Overrides Sub Execute(ByVal context As Context)
        If context.Tokens.Count > 0 Then
            context.Tokens.Add(context.Tokens.Last)
        End If
    End Sub

    Public Overrides Sub LoadCommand(ByVal project As Project, ByVal reader As System.IO.TextReader, ByVal line As String, ByRef lineNumber As Integer)
        MyBase.LoadCommand(project, reader, line, lineNumber)
        If line.ToLower() <> "repeat" Then
            project.Warnings.Add(String.Format("Line {0}: The repeat command requires 0 arguments.", lineNumber))
        End If
    End Sub

    Public Overrides Sub WriteCommand(ByVal writer As System.IO.TextWriter)
        writer.WriteLine("Repeat")
    End Sub

    Public Overrides Sub CheckSanity(ByVal project As Project)
    End Sub
End Class

File: monowordbuilder/wordbuilderbase/Commands/RepeatCommand.cs

namespace Whee.WordBuilder.Project.Commands
{
	[Highlight(RegEx = "^\\s*repeat$", Color = "Green")]
	public class RepeatCommand : CommandBase
	{
	
		public override void Execute(Context context)
		{
			if (context.Tokens.Count > 0) {
				context.Tokens.Add(context.Tokens[context.Tokens.Count - 1]);
			}
		}
	
		public override void LoadCommand(Project project, System.IO.TextReader reader, string line, ref int lineNumber)
		{
			base.LoadCommand(project, reader, line, ref lineNumber);
			if (line.ToLower() != "repeat") {
				project.Warnings.Add(string.Format("Line {0}: The repeat command requires 0 arguments.", lineNumber));
			}
		}
	
		public override void WriteCommand(System.IO.TextWriter writer)
		{
			writer.WriteLine("Repeat");
		}
	
		public override void CheckSanity(Project project)
		{
		}
	}
}